### PR TITLE
feat(make:factory): use factories to default non-nullable relationships

### PR DESCRIPTION
**Pull request by nikophil**
all work made by @benblub  :wink:

### CODE REVIEW

File: src/Bundle/Maker/MakeFactory.php [hunk 1]
```diff
@@ -268,6 +275,32 @@ private function defaultPropertiesForPersistedObject(string $class, bool $allFie
 
         $dbType = $em instanceof EntityManagerInterface ? 'ORM' : 'ODM';
 
+        // If Factory exist for related entities populate too with auto defaults
+        foreach ($metadata->associationMappings as $item) {
```
Comment (nikophil) [inline src/Bundle/Maker/MakeFactory.php:279]: I think I'll provide a more robust implementation in a further PR, it's time to split this class. I think we should handle `--all-fields` here as well

File: tests/Fixtures/Maker/expected/can_create_factory_with_relation_defaults.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    /**
+     * @see https://symfony.com/bundles/ZenstruckFoundryBundle/current/index.html#model-factories
+     *
+     * @todo add your default values here
+     */
+    protected function getDefaults(): array
+    {
+        return [
+            'manyToOne' => \Zenstruck\Foundry\Tests\Fixtures\Factories\CategoryFactory::new(),
```
Comment (kbond) [inline tests/Fixtures/Maker/expected/can_create_factory_with_relation_defaults.php:47]: Can we have this imported? I think a previous PR of yours played with the imports - can we easily hook into that?

Comment (nikophil) [reply tests/Fixtures/Maker/expected/can_create_factory_with_relation_defaults.php:47]: yep that's possible, but a little bit tricky: we return a `\Generator<string, string>` from `defaultPropertiesForPersistedObject()`:
```php
// generateFactory()
'defaultProperties' => $this->defaultPropertiesForPersistedObject(),

// defaultPropertiesForPersistedObject()
yield \lcfirst($fieldName) => "\\{$factoryClass}::new(),";
```

I think the cleanest solution would be to introduce some kind of "view model":
```php
class FactoryData
{
    private string $objectShortName;
    private array $uses;
    private array $defaultProperties;
    // ...
}
```

this way we could pass this instance to each functions which would modify it, until we give this object to the template which would use it to display all the needed informations.

WDYT?

Comment (kbond) [reply tests/Fixtures/Maker/expected/can_create_factory_with_relation_defaults.php:47]: Yes, that makes sense to me. Want to do that in this PR or another?

Comment (nikophil) [reply tests/Fixtures/Maker/expected/can_create_factory_with_relation_defaults.php:47]: I think a new PR will be better

**Pull request merged**